forgerock-auth/src/main/java/org/forgerock/android/auth/callback/WebAuthnAuthenticationCallback.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2022 ForgeRock. All rights reserved.
+ *
+ *  This software may be modified and distributed under the terms
+ *  of the MIT license. See the LICENSE file for details.
+ */
+package org.forgerock.android.auth.callback
+
+import android.annotation.TargetApi
+import android.content.Context
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.forgerock.android.auth.FRListener
+import org.forgerock.android.auth.Listener
+import org.forgerock.android.auth.Node
+import org.forgerock.android.auth.webauthn.WebAuthn
+import org.forgerock.android.auth.webauthn.WebAuthnAuthentication
+import org.forgerock.android.auth.webauthn.WebAuthnKeySelector
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * A callback that handle WebAuthnAuthentication Node.
+ */
+@TargetApi(24)
+open class WebAuthnAuthenticationCallback : MetadataCallback, WebAuthnCallback {
+
+    @JvmOverloads
+    constructor(jsonObject: JSONObject, index: Int) : super(jsonObject, index)
+
+    @JvmOverloads
+    constructor() : super()
+
+    override fun getType(): String {
+        return "WebAuthnAuthenticationCallback"
+    }
+
+    @Throws(Exception::class)
+    open fun getWebAuthnAuthentication() = WebAuthnAuthentication(value)
+
+    /**
+     * Perform WebAuthn Authentication
+     *
+     * @param context   The Application Context
+     * @param node      The Node returned from AM
+     * @param selector  Optional - The selector to select which credential key to use. Apply to Username-less only.
+     * @param listener  Listener to listen for result
+     */
+    fun authenticate(context: Context, node: Node,
```

Comment:
is there a timeout option do we need handle ?

Reply:
The timeout is handled by the Fido library:

 PublicKeyCredentialRequestOptions.Builder()
            .setAllowList(allowCredentials)
            .setRpId(relayingPartyId)
            .setChallenge(challenge)
            **.setTimeoutSeconds(timeout)**
            .build()